lib/src/widgets/custom_credit_card.dart

```diff
@@ -0,0 +1,238 @@
+    );
+  }
+
+  Widget _buildCvcFormField() {
+    return CardFormField(
+      inputDecoration: _buildInputDecoration(
+        hintText: widget.locale.cvc,
+      ),
+      inputFormatters: [
+        FilteringTextInputFormatter.digitsOnly,
+        LengthLimitingTextInputFormatter(4),
+      ],
+      validator: (String? input) => CardUtils.validateCVC(
+        input,
+        widget.locale,
+      ),
+      onChanged: _onCvcChange,
+    );
+  }
+
+  Widget _buildExpiryFormField() {
+    return CardFormField(
+      inputDecoration: _buildInputDecoration(
+        hintText: '${widget.locale.expiry} (MM / YY)',
+      ),
+      inputFormatters: [
+        FilteringTextInputFormatter.digitsOnly,
+        LengthLimitingTextInputFormatter(4),
+        CardMonthInputFormatter(),
+      ],
+      validator: (String? input) => CardUtils.validateDate(
+        input,
+        widget.locale,
+      ),
+      onChanged: _onExpiryDateChange,
+    );
+  }
+
+  void _onCvcChange(String? value) {
+    _cardData.cvc = value ?? '';
+    widget.onCreditCardFormChange(
+      _cardData,
+      _isValidForm(),
+    );
+  }
+
+  void _onExpiryDateChange(String? value) {
+    if (value != null) {
+      List<String> expireDate = CardUtils.getExpiryDate(value);
+      if (expireDate.length == 2) {
+        _cardData.month = expireDate.first;
+        _cardData.year = expireDate[1];
+      }
+      widget.onCreditCardFormChange(
+        _cardData,
+        _isValidForm(),
+      );
+    }
+  }
+
+  void _onCardNumberChange(String? value) {
+    if (value != null) {
+      _cardData.number = CardUtils.getCleanedNumber(value);
+      widget.onCreditCardFormChange(
+        _cardData,
+        _isValidForm(),
+      );
+    }
+  }
+
+  void _onNameOnCardChange(String? value) {
+    _cardData.name = value ?? '';
+    widget.onCreditCardFormChange(
+      _cardData,
+      _isValidForm(),
+    );
+  }
+
+  InputDecoration _buildInputDecoration({
+    required String hintText,
+    bool addNetworkIcons = false,
+  }) {
+    return InputDecoration(
+      suffixIcon: addNetworkIcons ? const NetworkIcons() : null,
+      hintText: hintText,
+      focusedErrorBorder: _defaultErrorBorder,
+      enabledBorder: _defaultEnabledBorder,
+      focusedBorder: _defaultFocusedBorder,
+      errorBorder: _defaultErrorBorder,
+    );
+  }
+
+  final OutlineInputBorder _defaultEnabledBorder = OutlineInputBorder(
+    borderSide: BorderSide(color: Colors.grey[400]!),
+    borderRadius: const BorderRadius.all(
+      Radius.circular(8),
+    ),
+  );
+
+  final OutlineInputBorder _defaultFocusedBorder = OutlineInputBorder(
+    borderSide: BorderSide(color: Colors.grey[600]!),
+    borderRadius: const BorderRadius.all(
+      Radius.circular(8),
+    ),
+  );
+
+  final OutlineInputBorder _defaultErrorBorder = const OutlineInputBorder(
+    borderSide: BorderSide(color: Colors.red),
+    borderRadius: BorderRadius.all(
+      Radius.circular(8),
+    ),
+  );
```

Comment:
Within a class, there is a certain order of members. Let's stick it out.
These variables should be in the variable area